packages/x-adapter-platform/src/mappers/request/search-request.mapper.ts

```diff
@@ -0,0 +1,50 @@
+import { createMutableSchema, Schema, schemaMapperFactory } from '@empathyco/x-adapter-next';
+import { reduce } from '@empathyco/x-utils';
+import { isHierarchicalFilter } from '@empathyco/x-types';
+import { PlatformSearchRequest, SearchRequest } from '../../types/request.types';
+
+export const searchRequestSchema: Schema<SearchRequest, PlatformSearchRequest> = {
+  device: 'device',
+  query: 'query',
+  env: 'env',
+  scope: 'scope',
+  origin: 'origin',
+  start: 'start',
+  rows: 'rows',
+  lang: 'lang',
+  sort: 'sort',
+  filter: mapFilters
+};
+
+export const searchRequestMutableSchema = createMutableSchema(searchRequestSchema);
+export const searchRequestMapper = schemaMapperFactory<SearchRequest, PlatformSearchRequest>(
+  searchRequestMutableSchema
+);
+
+/**
+ * Converts the filters to the shape the Platform's API is expecting.
+ *
+ * @param filters - The filters from our internal request.
+ *
```

Comment:
Should we write an example here? I know that we have the test, but maybe for readability.